### ISSUE TITLE
error compiling mexopencv from matlab

### ISSUE THREAD

**Issue by pedr0vieira** [kyamagu/mexopencv#148]
hi

I'm having trouble compiling mexopencv on ubunto 15.04 and matlab 2012b

this is the error

>> mexopencv.make
make MATLABDIR="/usr/local/MATLAB/R2012b" MEXEXT=mexa64  
/usr/local/MATLAB/R2012b/bin/mex -cxx -largeArrayDims -Iinclude -I/usr/local/include/opencv -I/usr/local/include  src/+cv/calcGlobalOrientation.cpp -lMxArray -Llib  -L/usr/local/lib/  -lopencv_calib3d  -lopencv_core  -lopencv_features2d  -lopencv_flann  -lopencv_hal  -lopencv_highgui  -lopencv_imgcodecs  -lopencv_imgproc  -lopencv_ml  -lopencv_objdetect  -lopencv_photo  -lopencv_shape  -lopencv_stitching  -lopencv_superres  -lopencv_ts  -lopencv_video  -lopencv_videoio  -lopencv_videostab -output +cv/calcGlobalOrientation
src/+cv/calcGlobalOrientation.cpp: In function ‘void mexFunction(int, mxArray**, int, const mxArray**)’:
src/+cv/calcGlobalOrientation.cpp:34: error: ‘calcGlobalOrientation’ was not declared in this scope

    mex: compile of ' "src/+cv/calcGlobalOrientation.cpp"' failed.

Makefile:37: recipe for target '+cv/calcGlobalOrientation.mexa64' failed
make: *** [+cv/calcGlobalOrientation.mexa64] Error 1

thanks

**Comment by kyamagu** [comment]
@pedr0vieira What is your OpenCV version? mexopencv is only compatible to OpenCV 2.4.x yet.

**Comment by pedr0vieira** [comment]
hi i'm using openCV 2.4.9 

anyway i managed to compile mexopencv, but now i get  this error

/usr/local/MATLAB/R2012b/bin/glnxa64/../../sys/os/glnxa64/libstdc++.so.6: version `GLIBCXX_3.4.15' not found (required by
/usr/local/lib/libopencv_calib3d.so.2.4)

=/

**Comment by amroamroamro** [comment]
@pedr0vieira:

>  -lopencv_shape -lopencv_hal -lopencv_imgcodecs -lopencv_videoio

These libs are definitely not in OpenCV 2.4.x, rather OpenCV 3.x (which is still not the stable release).

> /usr/local/MATLAB/R2012b/bin/glnxa64/../../sys/os/glnxa64/libstdc++.so.6:
> version `GLIBCXX_3.4.15' not found (required by /usr/local/lib/libopencv_calib3d.so.2.4)

see the [README file](https://github.com/kyamagu/mexopencv/blob/master/README.markdown#error-invalid-mex-file-or-segmentation-fault).

**Comment by pedr0vieira** [comment]
yes i know, the first time, i was using 3.x and ubuntu 15.04,  but now i'm using openCV 2.4.9  and ubuntu 14.04 (i find it more stable in my system)

tried what was mentioned in the readme file and now whenever i try to use a mex function , matlab crashes


------------------------------------------------------------------------
       Segmentation violation detected at Mon Apr 27 16:08:57 2015
------------------------------------------------------------------------

Configuration:
  Crash Decoding  : Disabled
  Current Visual  : 0x20 (class 4, depth 24)
  Default Encoding: UTF-8
  GNU C Library   : 2.19 stable
  MATLAB Root     : /usr/local/MATLAB/R2012b
  MATLAB Version  : 8.0.0.783 (R2012b)
  Operating System: Linux 3.16.0-30-generic #40~14.04.1-Ubuntu SMP Thu Jan 15 17:43:14 UTC 2015 x86_64
  Processor ID    : x86 Family 6 Model 60 Stepping 3, GenuineIntel
  Virtual Machine : Java 1.6.0_17-b04 with Sun Microsystems Inc. Java HotSpot(TM) 64-Bit Server VM mixed mode
  Window System   : The X.Org Foundation (11600000), display :0

Fault Count: 1


Abnormal termination:
Segmentation violation

Register State (from fault):
  RAX = 0000000000000000  RBX = 00007ff8f97197b0
  RCX = 00007ff99745e310  RDX = 00000000011f6090
  RSP = 00007ff99d7e0ac8  RBP = 0000000000000001
  RSI = 00007ffff5cfa8f8  RDI = 00007ff8f971f5f8

   R8 = 00007ff99747c580   R9 = 0000000000000040
  R10 = 00007ff99747bbd8  R11 = 00007ff9af7c0801
  R12 = 00007ffff5cfa8f8  R13 = 00000000011f6090
  R14 = 00007ff8f97197f8  R15 = 0000000000000000

  RIP = 000000000002e0a6  EFL = 0000000000010202

   CS = 0033   FS = 0000   GS = 0000

Stack Trace (from fault):


This error was detected while a MEX-file was running. If the MEX-file
is not an official MathWorks function, please examine its source code
for errors. Please consult the External Interfaces Guide for information
on debugging MEX-files.

If this problem is reproducible, please submit a Service Request via:
    http://www.mathworks.com/support/contact_us/

A technical support engineer might contact you with further information.

Thank you for your help.

**Comment by kyamagu** [comment]
@pedr0vieira You have to specify ALL conflicting dependencies, which typically include all OpenCV shared libs if you have Computer Vision Toolbox from Mathworks. Otherwise you will see segfault like this.

**Comment by pedr0vieira** [comment]
and how do i do that?

thanks for your patience

**Comment by kyamagu** [comment]
@pedr0vieira For example, if you're using Ubuntu's packaged OpenCV, you will launch matlab using something like the following. This will be of course different if you're using a custom-built OpenCV. Check the README again.

```bash
LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libstdc++.so.6:\
/usr/lib/x86_64-linux-gnu/libtiff.so:\
/usr/lib/x86_64-linux-gnu/libopencv_calib3d.so:\
/usr/lib/x86_64-linux-gnu/libopencv_contrib.so:\
/usr/lib/x86_64-linux-gnu/libopencv_core.so:\
/usr/lib/x86_64-linux-gnu/libopencv_features2d.so:\
/usr/lib/x86_64-linux-gnu/libopencv_flann.so:\
/usr/lib/x86_64-linux-gnu/libopencv_gpu.so:\
/usr/lib/x86_64-linux-gnu/libopencv_imgproc.so:\
/usr/lib/x86_64-linux-gnu/libopencv_legacy.so:\
/usr/lib/x86_64-linux-gnu/libopencv_ml.so:\
/usr/lib/x86_64-linux-gnu/libopencv_objdetect.so:\
/usr/lib/x86_64-linux-gnu/libopencv_photo.so:\
/usr/lib/x86_64-linux-gnu/libopencv_stitching.so:\
/usr/lib/x86_64-linux-gnu/libopencv_ts.so:\
/usr/lib/x86_64-linux-gnu/libopencv_video.so:\
/usr/lib/x86_64-linux-gnu/libopencv_videostab.so \
/usr/local/MATLAB/R2012b/bin/matlab
```
